ingestion_tools/scripts/transform_ingestion_configs.py

```diff
@@ -9,6 +9,32 @@ def cli():
     pass
 
 
+def create_deposition_metadata(deposition_id: str, config_data: dict[str, Any]) -> dict[str, Any]:
+    dataset = config_data.get("datasets", [])
+    metadata = dataset[0].get("metadata", {}) if dataset else {}
+    if metadata:
+        deposition_type = ["dataset"]
+    elif config_data.get("tomograms"):
+        metadata = config_data["tomograms"][0].get("metadata", {})
+        deposition_type = ["tomogram"]
+    elif config_data.get("annotations"):
+        metadata = config_data["annotations"][0].get("metadata", {})
+        deposition_type = ["annotations"]
+
+    return {
+        "deposition_identifier": deposition_id,
+        "deposition_title": "TBA",
+        "deposition_description": "TBA",
+        "deposition_type": deposition_type,
+        "dates": metadata.get("dates", {}),
+        "authors": metadata.get("authors", []),
+    }
+
+
+def has_no_sources(data: list[dict[str, Any]] | dict[str, Any]) -> bool:
+    return isinstance(data, dict) or not any(row.get("sources") for row in data)
+
+
 def update_config(data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
I think a lot of the code in this file is moot now since we've already translated old-style configs to new-style configs. 😓

Reply:
Gjensen configs still do the old style generation, and are updated to the new style with this script.

Reply:
ahhh ok. Maybe I'll fix the gjensen config script someday.